source/material_model/melt_simple.cc

```diff
@@ -110,115 +106,7 @@ namespace aspect
-                    double freezing_amount = std::max(porosity_change_wrt_melt_fraction, porosity_change_wrt_depletion);
-
-                    if (eq_melt_fraction == 0.0)
-                      freezing_amount = - old_porosity;
-
-                    porosity_change += freezing_amount;
-
-                    // Adapt time scale of freezing with respect to melting.
-                    // We have to multiply with the melting time scale here to obtain the porosity change
-                    // that happens in the time defined by the melting time scale (as opposed to a rate).
-                    // This is important because we want to perform some checks on this quantity (for example,
-                    // we want to make sure that this change does not lead to a negative porosity, see below).
-                    // Later on, the overall porosity change is then divided again by the melting time scale
-                    // to obtain the rate of melting or freezing, which is used in the operator splitting scheme.
-                    if (porosity_change < 0 )
-                      porosity_change *= freezing_rate * melting_time_scale;
-                  }
-                }
-
-              // remove melt that gets close to the surface
-              if (this->get_geometry_model().depth(in.position[i]) < extraction_depth)
-                porosity_change = -old_porosity * (in.position[i](1) - (this->get_geometry_model().maximal_depth() - extraction_depth))/extraction_depth;
-
-              // do not allow negative porosity
-              porosity_change = std::max(porosity_change, -old_porosity);
-
-              // because depletion is a volume-based, and not a mass-based property that is advected,
-              // additional scaling factors on the right hand side apply
-              for (unsigned int c=0; c<in.composition[i].size(); ++c)
-                {
-                  // fill reaction rate outputs
-                  if (reaction_rate_out != nullptr && in.requests_property(MaterialProperties::reaction_rates))
-                    {
-                      if (c == peridotite_idx && this->get_timestep_number() > 0)
-                        reaction_rate_out->reaction_rates[i][c] = porosity_change / melting_time_scale * (1 - maximum_melt_fraction) / (1 - old_porosity);
-                      else if (c == porosity_idx && this->get_timestep_number() > 0)
-                        reaction_rate_out->reaction_rates[i][c] = porosity_change / melting_time_scale;
-                      else
-                        reaction_rate_out->reaction_rates[i][c] = 0.0;
-                    }
-                  out.reaction_terms[i][c] = 0.0;
-                }
-
-              const double porosity = std::min(1.0, std::max(in.composition[i][porosity_idx],0.0));
-              out.viscosities[i] = eta_0 * exp(- alpha_phi * porosity);
```

Comment:
This line is now left out of the logic, such that `out.viscosities[i] = eta_0;` is the only option. I think this line should be retained in the revised structure?

Reply:
I think that's back in the logic with a change I just pushed. It's now in the main loop of  `calculate_fluid_outputs` and in `melt_simple` I left in the line `out.viscosities[i] *= visc_temperature_dependence;`. Does this look ok now? I might still be missing something here.